treeherder/webapp/api/jobs.py

```diff
@@ -91,10 +94,27 @@ def list(self, request, project, jm):
         - return_type (dict)
         """
         MAX_JOBS_COUNT = 2000
+        LAST_MODIFIED_WINDOW = 30  # minutes
+
         filter = UrlQueryFilter(request.query_params)
 
         offset = int(filter.pop("offset", 0))
         count = int(filter.pop("count", 10))
+        if "last_modified" in filter.conditions:
+            datestr = filter.get("last_modified")[1]
```

Comment:
Unfortunately `UrlQueryFilter()` (which is a homegrown thing from utils.py; I thought it was from d-r-f or d-r-f-filters initially) actually does a `.pop()` too, when you call `.get()`:
https://github.com/mozilla/treeherder/blob/6a5fbb731090f1723a0625ead7335832e591e9ba/treeherder/webapp/api/utils.py#L70

As such, whilst this PR successfully returns HTTP 400 for the bad cases we were trying to block, it also then stripped the `last_modified` attribute from all normal requests too, making them much more intensive.